src/PowerShellEditorServices/Hosting/HostStartupInfo.cs

```diff
@@ -158,6 +166,13 @@ public sealed class HostStartupInfo
         #endregion
     }
 
+    /// <summary>
+    /// This is a strange class that is generally <c>null</c> or otherwise just has a single path
+    /// set. It is eventually parsed one-by-one when setting up the PowerShell runspace.
```

Comment:
This class is used for configuring profile path values, which are set by the client. Different clients conceptually implement different hosts (so VSCode implements a different host to vim or IntelliJ), so they need different profile paths. This is how they configure that.